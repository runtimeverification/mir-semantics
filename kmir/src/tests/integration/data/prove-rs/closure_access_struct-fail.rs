struct MyStruct {
    data: i32
}

fn main() {
    // List of MyStruct instances
    let struct_list = [
        MyStruct { data: 10 },
        MyStruct { data: 20 },
        MyStruct { data: 30 },
        MyStruct { data: 40 },
        MyStruct { data: 50 }
    ];
    
    // Closure function that takes &MyStruct reference
    let get_value = |struct_ref: &MyStruct| {
        struct_ref.data
    };
    
    // Use list call style, passing reference
    let result = get_value(&struct_list[2]);
    
    // Verify result
    assert_eq!(result, 30);
}