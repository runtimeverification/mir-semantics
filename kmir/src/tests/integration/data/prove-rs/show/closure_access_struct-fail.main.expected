
┌─ 1 (root, init)
│   #execTerminator ( terminator ( ... kind: terminatorKindCall ( ... func: operandC
│   span: 0
│
│  (277 steps)
└─ 3 (stuck, leaf)
    #traverseProjection ( toLocal ( 19 ) , AllocRef ( allocId ( 0 ) , noMetadata ) ,
    function: main
    span: 106


┌─ 2 (root, leaf, target, terminal)
│   #EndProgram ~> .K


